pkgs/applications/audio/jamulus/default.nix

```diff
@@ -0,0 +1,36 @@
+{ mkDerivation, stdenv, fetchFromGitHub, pkgconfig, qtscript, qmake, libjack2
+}:
+
+mkDerivation {
+  pname = "jamulus";
+  version = "3.5.3";
+  src = fetchFromGitHub {
+    owner = "corrados";
+    repo = "jamulus";
+    rev = "5385f4f113884537c18109e01791149754fe96d5";
+    sha256 = "0wrg0g2cligk3pd82h2sh38xgfbdkwklinrk4p0f4iyfiqfzf0ck";
+  };
+
+  nativeBuildInputs = [ pkgconfig qmake ];
+  buildInputs = [ qtscript libjack2 ];
+
+  buildPhase = ''
+    make clean
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
```

Comment:
Same here, does the default `installPhase` not work?

Reply:
without this custom installPhase there didn't seem to be any binaries in the directory that was is printed in the last line of nix-build:
`/nix/store/8ajb9gy55ms8dfi7hljpz8xg342qggaj-jamulus-3.5.3
➜  nixpkgs git:(submit/jamulus) ✗ ls /nix/store/8ajb9gy55ms8dfi7hljpz8xg342qggaj-jamulus-3.5.3 
➜  nixpkgs git:(submit/jamulus) ✗ 
`

Reply:
Hmm, looks like there is no `install` target:

```
make: Nothing to be done for 'install'.
```

----

When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`.

---

Also the following will install the file and create necessary directories automatically:

```
install -Dt $out/bin Jamulus
```

Reply:
so {pre,post}Install are hooks that might not be used right now, but that every installPhase should call so someone could use them if they want?
In [another package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/generic-builder.nix) they seem to be simply called as first/last step of installPhase, so I did the same now

Reply:
I assume there there isn't a clean way to add an install target for make, so we just do `install -Dt $out/bin Jamulus`, right?

Reply:
Right, people can do `jamulus.overrideAttrs (attrs: { postInstall = ''foo''; })` and without `runHook`, they would need to override the whole `installPhase`.

Reply:
I have no experience with qmake but looking at their documentation, they apparently consider installing files an advanced feature:

https://doc.qt.io/qt-5/qmake-advanced-usage.html#installing-files

It would likely consist of adding something like:

https://github.com/NixOS/nixpkgs/blob/4b4e2b390ae53e1213081504b201a8e0e2e566f7/pkgs/applications/misc/qt-box-editor/default.nix#L26-L30

somewhere inside the unix section of the project file:

https://github.com/corrados/jamulus/blob/d1966d94232a37a7a9233c68d8d8ec3184040e98/Jamulus.pro#L264-L304

Reply:
hm I wonder if this would be somewhat brittle on future versions (unless we'd do some smart parsing of the Jamulus.pro file maybe), and might be less readable than the straightforward `install -Dt $out/bin Jamulus`... or is there some real advantage to be gained here?

Reply:
Sending the change as a pull request upstream would be best. No need for brittle text replacement, you can add the patch file directly (downloaded from the upstream commit using `fetchpatch`) to `patches` attribute.

Reply:
tried a couple of variants to make this install method (with patched Jamulus.pro instead of custom installPhase) work... 
setting 
`INSTALLS += target\ntarget.path = $out/bin`
results in the binary being installed to `.../source/ut/bin`, using braces like
`INSTALLS += target\ntarget.path = $(out)/bin`
installs to `.../source//nix/store/grqwx7liy5840kw6fpg9aph2fmd055kk-jamulus-3.5.3/bin/Jamulus` (so in this cases the expansion works the way I'd expect according to the qmake [docs](https://doc.qt.io/qt-5/qmake-language.html#variable-expansion) but apparently qmake doesn't treat it as an abspath... However, in case of `target.path = /tmp/myabspath` the path *is* treated as absolute

Reply:
Maybe try `$$PREFIX` like in the PR fix above, or `$$(out)`. Hopefully, then qmake will be able to see that it is absolute and not concatenate.

Reply:
`$$(out)` works! made a [PR](https://github.com/corrados/jamulus/pull/247) to jamulus, if it gets accepted we can use the patch and will only need `qmakeFlags = [ "INSTALL_PATH=$$(out)/bin" ];` instead of the custom `installPhase`

Reply:
Thanks for all the hints, with the jamulus PR merged it works with just fetchpatch and the default installPhase now